### PR TITLE
Added client logic, integration, UI, validations

### PR DESCRIPTION
**Pull request by Kitwradr**
Added api integration with text input

### CODE REVIEW

**Comment by Kitwradr** [comment]
https://github.com/Untellable/210-Team4-TDD/issues/4

**Comment by Kitwradr** [comment]
Basic integration with backend API and display of the relationships
Input text and button to update the graph as per user entry (with better styling)
Added validation so that user can only enter Numbers in text (an alert pops up if it does not match regex)
Added a spinner and overlay so that its evident the API call is in progress
Tried to make the website as responsive as possible - adapts to different screen sizes

**Review by alexluo1998 — CHANGES_REQUESTED** [review]
1. See my comments for an idea on what to unit test
2. Take a chance to review my PR so that the client node.js setup can be merged to main. And then `npm run ci` to run prettier and linter (should fix a lot of style issue I see here).
3. the new structure for our repo is 
/
/client/src
/server/src
make sure to fix it to avoid merge conflict after my PR is merged to main

**Review by alexluo1998 — APPROVED** [review]
lgtm! Add more unit tests later

File: client/src/process.js
Comment (Untellable) [inline client/src/process.js:49]: Move user ID verification to its own function?

File: client/src/process.js
Comment (Untellable) [inline client/src/process.js:95]: If this mainID default value is intentional, pls put a comment explaining what it means.

File: client/package.json
Comment (Untellable) [inline client/package.json:10]: Change this in server/package.json as well?

File: client/src/process.js
Comment (Untellable) [inline client/src/process.js:105]: Meaning of value is unclear here

**Review by Untellable — APPROVED** [review]
Noted a few minor opportunities for refactoring. Not sure if it's worth holding up the PR for that so up to you to merge and make the changes later or do it now.

Edit: Could you also add a bit to the readme (main readme and/or make a new one in /client like in /server) about how to run/test the client if it's any different than the server.

File: client/src/process.js
Comment (alexluo1998) [reply client/src/process.js:105]: the value is related to the size of the node, but actually we can try to remove it (and see if it still work) for this PR since that is another feature

File: client/src/process.js
Comment (alexluo1998) [reply client/src/process.js:95]: Nice catch - let's get rid of this default value and instead try to handle the case (log or show error) when id is not provided

File: client/src/process.js
Comment (alexluo1998) [reply client/src/process.js:49]: yeah and as I mentioned before, this can be a easy funtion to unit test

File: client/package.json
Comment (alexluo1998) [reply client/package.json:10]: Actually I will fix it in another PR (along with the "npm run start" script)

**Review by alexluo1998 — CHANGES_REQUESTED** [review]

File: client/src/process.js
Comment (Kitwradr) [reply client/src/process.js:95]: Good comment, resolving it

File: client/src/process.js
Comment (Kitwradr) [reply client/src/process.js:49]: Yep, sounds good

File: client/src/process.js
Comment (Kitwradr) [reply client/src/process.js:105]: Good comment, added comment explaining it's for size of node

**Review by alexluo1998 — APPROVED** [review]

File: client/src/process.js
Comment (Untellable) [reply client/src/process.js]: *Sorry, to clarify I meant creating a function which just verifies a userID. i.e. a function that accepts a string and returns true/false on whether it's valid. This would be a lot easier to unit test as you don't need to mock the html you can just test with a few different strings, and it can be reused if we need to verify the userID anywhere else. I think leaving the document.getElementById('idInput').value in the main function is fine bc. the function is specifically for the case of a user pressing the update button (and that way you only have to call it once).

File: client/src/process.js
Comment (Untellable) [reply client/src/process.js]: Possibly use the userID verify function mentioned in the previous comment to handle case of id not valid? May be unecessary if we can guarantee that the userID being passed is already verified, but probably safer to just handle it anyway.

Comment (Kitwradr) [reply client/src/process.js]: I think at this point, it would already be verified. Not sure if it would provide any value having it again

File: client/src/process.js
Comment (Kitwradr) [reply client/src/process.js]: Modified

**Pull request merged**
